docs/reference/search/search-your-data/semantic-search.asciidoc

```diff
@@ -0,0 +1,131 @@
+[[semantic-search]]
+== Semantic search
+
+Semantic search is a search method that helps you find data based on the intent
+and contextual meaning of a search query, instead of a match on query terms
+(lexical search).
+
+{es} provides semantic search capabilities using {ml-docs}/ml-nlp.html[natural
+language processing (NLP)] and vector search. Deploying an NLP model to {es}
+enables it to extract text embeddings out of text. Embeddings are vectors that
+provide a numeric representation of a text. Pieces of content with similar
+meaning have similar representations. 
+
+[[semantic-search-diagram]]
+.A simplified representation of encoding textual concepts as vectors
+image::images/search/vector-search-oversimplification.png[A simplified representation of encoding textual concepts as vectors,align="center"]
+
+At query time, {es} can use the same NLP model to convert a query into
+embeddings, enabling you to find documents with similar text embeddings.
+
+This guide shows you how to implement semantic search with {es}, from selecting
+an NLP model, to writing queries.
+
+[discrete]
+[[semantic-search-select-nlp-model]]
+=== Select an NLP model
+
+{es} offers the usage of a 
+{ml-docs}/ml-nlp-model-ref.html#ml-nlp-model-ref-text-embedding[wide range of NLP models], 
+including both dense and sparse vector models. Your choice of the language model 
+is critical for implementing semantic search successfully.
+
+While it is possible to bring your own text embedding model, achieving good 
+search results through model tuning is challenging. Selecting an appropriate 
+model from our third-party model list is the first step. Training the model on 
+your own data is essential to ensure better search results than using only BM25. 
+However, the model training process requires a team of data scientists and ML 
+experts, making it expensive and time-consuming.
+
+To address this issue, Elastic provides a pre-trained representational model 
+called {ml-docs}/ml-nlp-elser.html[Elastic Learned Sparse EncodeR (ELSER)]. 
+ELSER, currently available only for English, is an out-of-domain sparse vector 
+model that does not require fine-tuning. This adaptability makes it suitable for 
+various NLP use cases out of the box. Unless you have a team of ML specialists, 
+it is highly recommended to use the ELSER model.
+
+In the case of sparse vector representation, the vectors mostly consist of zero 
+values, with only a small subset containing non-zero values. This representation 
+is commonly used for textual data. In the case of ELSER, each document in an 
+index and the query text itself are represented by high-dimensional sparse 
+vectors. The interaction between these large sparse vectors can be very 
+sophisticated, as the dot product of those vectors provides greater details and 
+interpretability compared to dense vectors. This sophisticated interaction 
+enables ELSER to efficiently capture context.
```

Comment:
I am not sure this is the best way to describe it to non-experts. Maybe it would be better to focus on the text expansion aspect:

```suggestion
vectors. Each non-zero element of the vector corresponds to a term in the model vocabulary.
The ELSER vocabulary contains around 30000 terms, so the sparse vectors created by ELSER
contain about 30000 values, the majority of which are zero. Effectively the ELSER model is
replacing the terms in the original query with other terms that have been learnt to exist in the
documents that best match the original search terms in a training dataset, and weights to
control how important each is.
```